src/App.jsx

```diff
@@ -14,45 +15,185 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
+const goodsWithId = goodsFromServer.map(good => (
+  {
+    name: good,
+    // eslint-disable-next-line
+    id: self.crypto.randomUUID(),
+  }
+));
+
+let renderedGoods;
+
+const SORT_VALUE_NAME = 'name';
+const SORT_VALUE_LENGTH = 'length';
+let SORT_VALUE_REVERSE = 'reverse';
+let isReversed = false;
+let isAlpha = false;
+let isLength = false;
+
+function getPreperedGoods(goods = goodsWithId, sortField, reverse) {
+  let preperedGoods = [...goods];
+
+  if (sortField !== SORT_VALUE_REVERSE) {
+    preperedGoods.sort((good, good2) => {
+      if (sortField === 'length') {
+        isLength = true;
+        isAlpha = false;
+        if (!isReversed) {
+          return good.name[sortField] - good2.name[sortField];
+        // eslint-disable-next-line
+        } else {
+          return good2.name[sortField] - good.name[sortField];
+        }
+      }
+
+      if (sortField === 'name') {
+        isAlpha = true;
+        isLength = false;
+
+        return good[sortField].localeCompare(good2[sortField]);
+      }
+
+      return 0;
+    });
+  }
+
+  if ((sortField === SORT_VALUE_REVERSE || isReversed)
+      && sortField !== SORT_VALUE_LENGTH) {
+    if (sortField === SORT_VALUE_REVERSE) {
+      isReversed = !isReversed;
+    }
+
+    SORT_VALUE_REVERSE += ' ';
+    preperedGoods.reverse();
+  }
+
+  if (sortField === 'reset') {
+    isAlpha = false;
+    isReversed = false;
+    isLength = false;
+    preperedGoods = [...goodsWithId];
+  }
+
+  renderedGoods = preperedGoods;
+
+  return preperedGoods;
+}
+
+export const App = () => {
+  const [sortValue, setSortValue] = useState('');
+  const [reverseArray, setReverseArray] = useState(false);
```

Comment:
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig

Tips on naming boolean variables